app/controllers/service_providers_controller.rb

```diff
@@ -78,9 +78,10 @@ def publish
   private
 
   def help_text_empty?
-    service_provider.help_text['sign_in'].blank? &&
-      service_provider.help_text['sign_up'].blank? &&
-      service_provider.help_text['forgot_password'].blank?
+    service_provider.
+      help_text.
+      values.
+      all? {|text| text.values.all?('')}
```

Comment:
I'd keep the `.blank?` if possible, because things like `"    ".blank? == true`
```suggestion
      all? { |text| text.values.all?(&:blank) }
```

Reply:
oh yah good idea!

Reply:
i moved it to a view helper so i could more easily test (also since it's just for view-purposes that makes more sense there, to me anyways)

https://github.com/18F/identity-dashboard/pull/777/commits/6a48c4b3db0f459d13463e159ff72b671b95f7a5